src/DfuError.js

```diff
@@ -0,0 +1,193 @@
+const debug = require('debug')('dfu:error');
+
+export const ErrorCode = {
+    // Error message types
+    ERROR_MESSAGE: 0x00,
+    ERROR_MESSAGE_RSP: 0x01,
+    ERROR_MESSAGE_EXT: 0x02,
+
+    // Error code for DfuAbstractTransport
+    ERROR_CAN_NOT_INIT_ABSTRACT_TRANSPORT: 0x0000,
+    ERROR_PRE_DFU_INTERRUPTED: 0x0001,
+    ERROR_UNEXPECTED_BYTES: 0x0002,
+    ERROR_CRC_MISMATCH: 0x0003,
+    ERROR_TOO_MANY_WRITE_FAILURES: 0x0004,
+
+    // Error code for DfuTransportPrn
+    ERROR_CAN_NOT_INIT_PRN_TRANSPORT: 0x0010,
+    ERROR_CAN_NOT_USE_HIGHER_PRN: 0x0011,
+    ERROR_READ_CONFLICT: 0x0012,
+    ERROR_TIMEOUT_READING_SERIAL: 0x0013,
+    ERROR_RECEIVE_TWO_MESSAGES: 0x0014,
+    ERROR_RESPONSE_NOT_START_WITH_0x60: 0x0015,
+    ERROR_ASSERT_EMPTY_RESPONSE: 0x0016,
+    ERROR_UNEXPECTED_RESPONSE_OPCODE: 0x0017,
+    ERROR_UNEXPECTED_RESPONSE_BYTES: 0x0018,
+
+    // Error code for DfuTransportSink
+    ERROR_MUST_HAVE_PAYLOAD: 0x0031,
+    ERROR_INVOKED_MISMATCHED_CRC32: 0x0032,
+    ERROR_MORE_BYTES_THAN_CHUNK_SIZE: 0x0033,
+    ERROR_INVALID_PAYLOAD_TYPE: 0x0034,
+
+    // Error code for DfuTransportNoble
+    ERROR_CAN_NOT_DISCOVER_DFU_CONTROL: 0x0051,
+    ERROR_TIMEOUT_FETCHING_CHARACTERISTICS: 0x0052,
+    ERROR_CAN_NOT_SUBSCRIBE_CHANGES: 0x0053,
+
+    // Error code for DfuTransportSerial(including slow and usb)
+    ERROR_UNKNOWN_FIRMWARE_TYPE: 0x0071,
+    ERROR_UNABLE_FIND_PORT: 0x0072,
+
+    // Error code for response error messages
+    ERROR_RSP_OPCODE_MISSING_MALFORMED: 0x0100,
```

Comment:
For ERROR_RSP and ERROR_EXT there exists definitions in the SDK. I suggest reusing those wordings for consistency (minus the namespace the use in front):
[SDK DFU result error codes](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/group__nrf__dfu__rescodes.html#ga50daf693717ee90d08614d2a337e2361)
[SDK DFU extended error codes](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/group__sdk__nrf__dfu__req__handler.html#ga654d8446f2996253016f7c7713124094)

Reply:
Good point. I will follow the wordings there.